File: backend/IdentityManager/Models/Connector.cs
namespace IdentityManager.Models;

public class Connector
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty; // e.g. Active Directory, SAP, GitHub
    public string Type { get; set; } = string.Empty; // e.g. Direct, Web Service, Flat File
    public string Host { get; set; } = string.Empty;
    public string Status { get; set; } = "Healthy";
    public List<Application> Applications { get; set; } = new();
    public string ProvisioningConfig { get; set; } = "{}"; // JSON config for provisioning fields
}

File: frontend/identity-ui/src/app/services/connector.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Application {
  id: number;
  name: string;
  description: string;
  isTrustedSource: boolean;
}

export interface Connector {
  id: number;
  name: string;
  type: string;
  host: string;
  status: string;
  provisioningConfig: string;
  applications: Application[];
}

@Injectable({
  providedIn: 'root'
})
export class ConnectorService {
  private apiUrl = 'http://localhost:5283/api'; // Adjusted to match backend port

  constructor(private http: HttpClient) { }

  getConnectors(): Observable<Connector[]> {
    return this.http.get<Connector[]>(`${this.apiUrl}/connectors`);
  }

  getConnector(id: number): Observable<Connector> {
    return this.http.get<Connector>(`${this.apiUrl}/connectors/${id}`);
  }

  updateProvisioning(id: number, config: string): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/connectors/${id}/provisioning`, `"${config}"`, {
      headers: { 'Content-Type': 'application/json' }
    });
  }

  getTrustedSources(): Observable<Application[]> {
    return this.http.get<Application[]>(`${this.apiUrl}/sources`);
  }

  aggregate(id: number): Observable<any> {
    return this.http.post(`${this.apiUrl}/sources/${id}/aggregate`, {});
  }
}
